packages/smartcontracts/src/tests/BridgeProxy.proxyBehaviour.spec.ts

```diff
@@ -0,0 +1,151 @@
+      expect(await upgradedProxyBridge.transactionFee()).to.equal(30);
+    });
+  });
+
+  describe('Test different situations of calling the proxy contract', () => {
+    let abiCoder: any;
+    let proxyBridge: BridgeV1;
+    let defaultAdminSigner: SignerWithAddress;
+
+    beforeEach(async () => {
+      abiCoder = new ethers.utils.AbiCoder();
+      const bridgeDeploymentResult: BridgeDeploymentResult = await loadFixture(deployContracts);
+      proxyBridge = bridgeDeploymentResult.proxyBridge;
+      defaultAdminSigner = bridgeDeploymentResult.defaultAdminSigner;
+    });
+
+    it('Send no data and send some ether to the smart contract', async () => {
+      const prevETHBalance = await ethers.provider.getBalance(proxyBridge.address);
+      const tx = await defaultAdminSigner.sendTransaction({
+        to: proxyBridge.address,
+        data: '0x',
+        value: toWei('1'),
+      });
+      const receipt = await tx.wait();
+      // check the event emitted
+      expect(receipt.logs[0].address).to.equal(proxyBridge.address);
+      expect(receipt.logs[0].topics[0]).to.equal(
+        ethers.utils.keccak256(ethers.utils.toUtf8Bytes('ETH_RECEIVED(address,uint256)')),
+      );
+      expect(receipt.logs[0].topics[1]).to.equal(abiCoder.encode(['address'], [defaultAdminSigner.address]));
+      expect(receipt.logs[0].topics[2]).to.equal(abiCoder.encode(['uint256'], [toWei('1')]));
+      const afterETHBalance = await ethers.provider.getBalance(proxyBridge.address);
+      expect(afterETHBalance.sub(prevETHBalance)).to.equal(toWei('1'));
+    });
+
+    it('Call a method that is not in the implementation interface', async () => {
+      const TestTokenFactory = await ethers.getContractFactory('TestToken');
+      const attachedToken = TestTokenFactory.attach(proxyBridge.address);
+      await expect(attachedToken.totalSupply()).to.reverted;
+    });
+
+    it('Call a method that is in the implementation interface that is not payable, without sending ether', async () => {
+      await defaultAdminSigner.sendTransaction({
+        to: proxyBridge.address,
+        data:
+          ethers.utils.keccak256(ethers.utils.toUtf8Bytes('addSupportedTokens(address,uint256)')).substring(0, 10) +
+          abiCoder.encode(['address', 'uint256'], [ethers.constants.AddressZero, toWei('10')]).substring(2),
+      });
+      expect(await proxyBridge.supportedTokens(ethers.constants.AddressZero)).to.equal(true);
+    });
+
+    it('Call a method that is in the implementation interface that is not payable, and send ether', async () => {
+      await expect(
+        defaultAdminSigner.sendTransaction({
+          to: proxyBridge.address,
+          data:
+            ethers.utils.keccak256(ethers.utils.toUtf8Bytes('addSupportedTokens(address,uint256)')).substring(0, 10) +
+            abiCoder.encode(['address', 'uint256'], [ethers.constants.AddressZero, toWei('10')]).substring(2),
+          value: 100,
+        }),
+      ).to.reverted;
```

Comment:
```suggestion
      ).to.be.reverted;
```
Is it this instead? 🤔